### PR TITLE
dhcp: T2265: refactor DHCP class

### PR DESCRIPTION
**Pull request by thomas-mangin**
Breaks the code down to make it easier to understand, simplify API

### CODE REVIEW

File: python/vyos/ifconfig/dhcp.py [hunk 1]
```diff
@@ -16,103 +16,55 @@
...
-        """
-        return self._dhcp[6]['options']
-
-    def set_dhcpv6_options(self, options):
-        """
-        Store new DHCP options used by next run of DHCP client.
-        """
-        self._dhcp[6]['options'] = options
+class _DHCPv4 (_DHCP):
+    template = """\
```
Comment (c-po) [inline python/vyos/ifconfig/dhcp.py]: Please migrate the template to data/templates folder

**Review by c-po — CHANGES_REQUESTED** [review]
Please migrate the templates to data/templates folder

File: python/vyos/ifconfig/dhcp.py (hunk 1 above)
Comment (thomas-mangin) [reply python/vyos/ifconfig/dhcp.py]: Hi Christian, I agree it should be done but could you please merge as the current code does not load them from there neither so it is not a step backward.

Comment (c-po) [reply python/vyos/ifconfig/dhcp.py]: Can‘t as there are conflicting files.

Comment (thomas-mangin) [reply python/vyos/ifconfig/dhcp.py]: argh ! ok - will re,re,...base :-) and add the template at the same time.

**Comment by thomas-mangin** [comment]
migration to templated added to the patchset.

**Pull request merged**